File: smonitor-web/src/main/java/org/lorislab/smonitor/admin/client/model/AgentWrapper.java
/*
 * Copyright 2013 lorislab.org.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.lorislab.smonitor.admin.client.model;

import org.lorislab.smonitor.rs.admin.model.Agent;
import org.lorislab.smonitor.rs.model.ServerInfo;

/**
 *
 * @author Andrej Petras
 */
public class AgentWrapper {
    
    public Agent agent;
    
    public ServerInfo server;
        
    public boolean connected;
    
    public String error;
    
    public void clear() {
        server = null;
        connected = false;
        error = null;
    }
}

File: smonitor-web/src/main/webapp/assets/js/org/lorislab/smonitor/wrapper/AgentWrapper.js
/**
 * The agent wrapper.
 * 
 * @constructor
 * @returns {AgentWrapper} the new created agent.
 * @param {org.lorislab.smonitor.rs.model.Agent} data the server side object server info.
 * 
 * @see org.lorislab.smonitor.rs.model.Agent
 * 
 * @author Andrej Petras
 */
function AgentWrapper(data) {
	this.agent = new Agent();
	this.agent.set(data);
	this.serverInfo = null;
	this.request = ko.observable(false);
	this.connected = ko.observable(false);
	this.message = ko.observable(null);
}

AgentWrapper.prototype.update = function(data) {
	this.agent.set(data);
};

AgentWrapper.prototype.statusOk = function(data) {
	if (this.serverInfo != null) {
		this.serverInfo.updateApp(data);
	} else {
		this.serverInfo = new ServerInfo(data);
	}
	this.request(false);
	this.message(null);
	this.connected(true);
};

AgentWrapper.prototype.statusError = function(data) {
	this.serverInfo = null;
	this.request(false);
	if (data != null) {
		this.message(data.message);
	} else {
		this.message(null);
	}
	this.connected(false);
};

AgentWrapper.prototype.statusStart = function() {
	this.serverInfo = null;
	this.request(true);
	this.message(null);
	this.connected(false);
};